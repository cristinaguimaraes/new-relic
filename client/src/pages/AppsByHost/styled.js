import styled from 'styled-components';

export const Content= styled.div`
width: 840px;
display:flex;
justify-content: center;
padding: 30px;
flex-direction: column;
background-color: #F8F8F8
`;

export const Cards= styled.div`
display:flex;
flex-wrap: wrap;
width: 840px;
`;


