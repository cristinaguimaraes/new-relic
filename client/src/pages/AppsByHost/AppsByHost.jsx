import React, {  useState } from 'react';
import {Card} from '../../components/Card';
import {Content,  Cards} from './styled';
import { arrayOf, number, shape, string } from 'prop-types';
import {AppsByHostHeader} from "../../components/AppsByHostHeader";
import {Redirect} from "react-router-dom";

const AppsByHost = ({data}) => {

    const [layout, setLayout] = useState('list');

    if (Object.keys(data).length === 0) return <Redirect to="/" />;
    
    const handleOnChangeLayoutClick = () => {
        setLayout(prevState => prevState === 'list' ? 'grid' : 'list')
    };

    return (
        <Content>
            <AppsByHostHeader onClick={handleOnChangeLayoutClick} layout={layout}/>
            <Cards>
                {Object.entries(data).map(([host, apps], index) => (
                        <Card key={`host-${host}`} index={index} layout={layout} host={host} apps={apps.slice(0, 5)}/>
                    ))
                }
            </Cards>
            
        </Content>
    )
};

AppsByHost.propTypes = {
    location: shape({state:shape({hostsWithApps:arrayOf(shape({apdex: number.isRequired, name: string.isRequired})).isRequired})}).isRequired,
};

export { AppsByHost };
