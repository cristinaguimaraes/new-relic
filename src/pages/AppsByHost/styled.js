import styled from "styled-components";

export const Content = styled.div`
  width: 840px;
  display: flex;
  justify-content: center;
  padding: 30px;
  flex-direction: column;
  background-color: #f8f8f8;
`;

export const Cards = styled.div`
  display: flex;
  flex-wrap: wrap;
  width: 840px;
`;
