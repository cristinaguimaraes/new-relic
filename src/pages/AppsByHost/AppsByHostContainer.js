import { AppsByHost } from "./AppsByHost";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { hostsWithApps } from "../../redux/reducers/apps";

const mapStateToProps = state => ({
  data: hostsWithApps(state.apps)
});

export const AppsByHostContainer = withRouter(
  connect(mapStateToProps, null)(AppsByHost)
);
