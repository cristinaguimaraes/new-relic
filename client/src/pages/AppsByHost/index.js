import { AppsByHostContainer } from './AppsByHostContainer';
export { AppsByHostContainer as AppsByHost };
