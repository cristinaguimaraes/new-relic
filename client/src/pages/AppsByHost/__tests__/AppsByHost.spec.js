import { AppsByHost} from '../AppsByHost';
import { shallow } from 'enzyme';
import React from 'react';

describe('AppsByHost component', () => {
  test('should render properly', () => {
    expect(shallow(<AppsByHost >Test</AppsByHost>)).toMatchSnapshot();
  });
});
