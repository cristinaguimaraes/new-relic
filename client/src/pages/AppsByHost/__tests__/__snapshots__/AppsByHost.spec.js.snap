// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppsByHost component should render properly 1`] = `ShallowWrapper {}`;
